tests/main/interfaces-ros-snapd-support/task.yaml

```diff
@@ -0,0 +1,49 @@
+summary: Ensure that the ros-snapd-support interface works.
+
+details: |
+    The ros-snapd-support interface allows the ros-snapd snap to
+    control snap applications.
+
+    The ros-snapd and ros2-snapd snaps define the ros-snapd-support plug.
+
+    The snaps declaring a plug on this interface must be able to control
+    snap applications through the snapd-snap.socket.
+    The test snap used has a command to get the list of snap applications
+    (exercising the read capability on the socket) and a command stop a snap application
+    (which checks the write capability).
+    #TODO A network plug must
+    be defined and connected for the snap to be able to talk to the socket, the
+    ros-snapd-support is not enough by itself.
+
+environment:
+    # not all terminals support UTF-8, but Python tries to be smart and attempts
+    # to guess the encoding as if the output would go to the terminal, but in
+    # fact all the test does is pipe the output to gojq
+    PYTHONIOENCODING: utf-8
+
+execute: |
+    "$TESTSTOOLS"/snaps-state install-local api-apps-client
+    echo "The ros-snapd-support plugs on the api-app-sclient snap is initially disconnected"
+    snap connections api-apps-client | MATCH "ros-snapd-support +api-apps-client:ros-snapd-support +- +-"
+
```

Comment:
you can also check here that the api-apps-client fails with 403

Reply:
Done in https://github.com/canonical/snapd/pull/15300/commits/42631d34070e23d61107441e5094726231508991